lib/tools/system-calls.js

```diff
@@ -608,4 +613,28 @@ systemCallMethods.ls = async function (remotePath) {
   }
 };
 
+systemCallMethods.fileSize = async function (remotePath) {
+  try {
+    let stdout = await this.shell(['ls', '-la', remotePath]);
+    let lines = stdout.split('\n');
+    lines = lines.map((l) => l.trim())
+      .filter(Boolean)
+      .filter((l) => l.indexOf('No such file') === -1);
+    if (lines.length !== 1) {
+      throw new Error(`Remote path is not a file`);
+    }
+    // https://regex101.com/r/fOs4P4/1
+    let match = /[-rwxd]{10}\s*\S*\s*\S*\s*\S*\s*(\d*)\s*\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}\s[\S\.]*/.exec(lines[0]);
+    if (!match || _.isNaN(parseInt(match[1], 10))) {
+      throw new Error(`Unable to parse size from list output: '${lines[0]}'`);
+    }
+    return parseInt(match[1], 10);
+  } catch (err) {
+    if (err.message.indexOf('No such file or directory') === -1) {
```

Comment:
Can we somehow improve the flow where we throw and catch custom errors? Probably, replace them with if conditions where possible?